src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -1,44 +1,104 @@
-cd "$SOURCE_DIR"
-
 # Yarn config is per user, and since the build might run with a non-root account, we make sure
 # the yarn cache is set on every build.
 YARN_CACHE_DIR=/usr/local/share/yarn-cache
 if [ -d $YARN_CACHE_DIR ]
 then
 	echo
-    echo "Configuring Yarn cache folder ..."
+    echo "Configuring Yarn cache folder..."
     yarn config set cache-folder $YARN_CACHE_DIR
 fi
 
+allNodeModulesDirName=__oryx_all_node_modules
+if [ -d "$SOURCE_DIR"/$allNodeModulesDirName ]
+then
+	# Clean-up any earlier build artifacts (which contain only prod dependencies)
+	rm -rf node_modules
+	rm -f node_modules.zip
+
+	# Rename the folder which has all the node modules to reuse for later builds to improve perf
+	mv "$SOURCE_DIR"/$allNodeModulesDirName "$SOURCE_DIR"/node_modules
```

Comment:
this is confusing... why would we have node_modules inside a node_modules folder?

Reply:
This is just renaming the folder